### PR TITLE
Object Storage (LEP 20230430)

### PR DESCRIPTION
**Pull request by m-ildefons**
User interface for managing the object stores of Longhorn.
This adds a new tab to the Longhorn UI, which is dedicated to managing object stores. It can list, create and delete object stores as well as show their operational status.
Each object store corresponds to an `ObjectStore` CRD in the K8s cluster, the Longhorn Manager implements a corresponding controller which orchestrates an [s3gw](https://github.com/aquarist-labs/s3gw) instance to implement that object store.

Related: https://github.com/longhorn/longhorn-manager/pull/2136
Related: https://github.com/longhorn/longhorn/pull/6640

LEP: https://github.com/longhorn/longhorn/pull/5832

### CODE REVIEW

File: src/models/objectStore.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            payload: location.pathname.startsWith('/objectstore') ? queryString.parse(location.search) : {},
+          })
+        }
+      })
+    },
+  },
+  effects: {
+    *query({ payload }, { call, put }) {
+      const data = yield call(listObjectStores, payload)
+      yield put({ type: 'listObjectStores', payload: { ...data } })
```
Comment (votdev) [inline src/models/objectStore.js:30]: Is a `put` used by reason here? I assume it should be `get` instead.

**Review by votdev — CHANGES_REQUESTED** [review]

File: src/models/objectStore.js (hunk 1 above)
Comment (m-ildefons) [reply src/models/objectStore.js:30]: It should be a `put`. `get` would cause errors

**Review by votdev — APPROVED** [review]

**Comment by PhanLe1010** [comment]
Hey, in the current version, the access key is keep changing when user are typing the ObjectStore name in UI. I am guessing it is hashing the volume of the ObjectStore in realtime. This is bit strange as other people pointed out. Can we just generate a random string and stick with it?

**Comment by PhanLe1010** [comment]
I couldn't figureout why we cannot access the s3gw-ui using your latest version at https://github.com/m-ildefons/s3gw-ui/tree/wip/longhorn-integration. Can you help?

Screenshot from 2023-11-20 17-03-18

**Comment by m-ildefons** [comment]
This is odd. I've never seen that error, maybe @votdev has an idea where it comes from. Can you provide a reproducer or some kind of hint what your configuration looks like?

**Pull request closed without merging**